app/assets/v2/js/cart-ethereum-zksync.js

```diff
@@ -0,0 +1,273 @@
+    async setupZkSync() {
+      this.user.address = (await web3.eth.getAccounts())[0];
+      this.zksync.checkoutManager = new ZkSyncCheckout.CheckoutManager(this.network || 'mainnet');
+      this.user.zksyncState = await this.zksync.checkoutManager.getState(this.user.address);
+    },
+
+    // Send a batch transfer based on donation inputs
+    async checkoutWithZksync() {
+      try {
+        // Send user to zkSync to complete checkout
+        this.zksync.checkoutStatus = 'pending';
+        const txHashes = await this.zksync.checkoutManager.zkSyncBatchCheckout(
+          this.transfers,
+          this.zksync.feeTokenSymbol
+        );
+
+
+        // Save contributors to database and redirect to success modal
+        _alert('Saving contributions. Please do not leave this page.', 'success', 2000);
+        // TODO save contributions
+        console.log('txHashes: ', txHashes);
+        await appCart.$refs.cart.finalizeCheckout(); // Update UI and redirect
+
+      } catch (e) {
+        switch (e) {
+          case 'User closed zkSync':
+            _alert('Checkout not complete: User closed the zkSync page. Please try again', 'error');
+            this.resetZkSyncModal();
+            throw e;
+
+          case 'Failed to open zkSync page':
+            _alert('Checkout not complete: Unable to open the zkSync page. Please try again', 'error');
+            this.resetZkSyncModal();
+            throw e;
+
+          case 'Took too long for the zkSync page to open':
+            _alert('Checkout not complete: Took too long to open the zkSync page. Please try again', 'error');
+            this.resetZkSyncModal();
+            throw e;
+
+          default:
+            this.handleError(e);
+        }
+      }
+    },
+
+    // Estimates the total gas cost of a zkSync checkout and sends it to cart.js
+    estimateGasCost() {
+      // Estimate minimum gas cost based on 400 gas per transfer
+      const gasPerTransfer = toBigNumber('400');
+      const numberOfTransfers = String(this.donationInputs.length);
+      const minimumCost = gasPerTransfer.mul(numberOfTransfers);
+      let totalCost = minimumCost;
+      
+      // If user has enough balance within zkSync, cost equals the minimum amount
+      const { isBalanceSufficient, requiredAmounts } = this.hasEnoughBalanceInZkSync();
+
+      if (isBalanceSufficient) {
+        return totalCost.toString();
+      }
+
+      // If we're here, user needs at least one L1 deposit, so let's calculate the total cost
+      this.cart.tokenList.forEach((tokenSymbol) => {
+        const zksyncBalance = toBigNumber(this.user.zksyncState.committed.balances[tokenSymbol]);
+        
+        if (requiredAmounts[tokenSymbol].gt(zksyncBalance)) {
+          if (tokenSymbol === 'ETH') {
+            totalCost = totalCost.add('200000'); // add 200k gas for ETH deposits
+          } else {
+            totalCost = totalCost.add('250000'); // add 250k gas for token deposits
```

Comment:
I believe zkSync uses 200k and 300k as the gas limits respectively, but since we aren't sending the transaction ourselves anymore we probably can be a bit less conservative here. I'll make a note of this and perhaps we circle back later once the rest of the core changes are made?

Reply:
Fine with me!